src/sidebar/components/Annotation/AnnotationEditor.tsx

```diff
@@ -178,6 +179,28 @@ function AnnotationEditor({
   const mentionsEnabled = store.isFeatureEnabled('at_mentions');
   const usersWhoAnnotated = store.usersWhoAnnotated();
   const focusedGroupMembers = store.getFocusedGroupMembers();
+  const usersForMentions = useMemo(() => {
+    // If group members are not loaded yet, we don't want to display any user.
+    // That will prevent a mix of some already-known users and a loading
+    // indicator showing at the same time in the suggestions popover.
+    if (!Array.isArray(focusedGroupMembers)) {
+      return 'loading';
+    }
+
+    // Once group members are loaded, we can merge them with the users who
+    // already annotated the document, then deduplicate and sort the result.
+    const focusedGroupUsers: UserItem[] = focusedGroupMembers.map(
```

Comment:
I would suggest extracting the logic for combining (merging, sorting) users from different sources into an external helper which can be tested on its own.